models/repo.go

```diff
@@ -1583,14 +1584,22 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 	}
 
 	repos = make([]*Repository, 0, opts.PageSize)
-
-	// Append conditions
-	sess := x.Where("LOWER(lower_name) LIKE ?", "%"+opts.Keyword+"%")
-	if opts.OwnerID > 0 {
-		sess.And("owner_id = ?", opts.OwnerID)
+	sess := x.Alias("repo")
+	// Attempt to find repositories that opts.UserId has access to
+	// This does not include other people's private repositories even if opts.UserId is an admin
+	if !opts.Private && opts.UserID > 0 {
+		sess.Join("LEFT", []string{"access", "acc"}, "acc.repo_id = repo.id")
```

Comment:
Yes. An inner join wouldn't work here since that would cause the query only to return repositories that had a record in the access table which would exclude public repositories and the users own private repositories. A left join here allows us to return all repositories that match the search string and one or more of the 3 optional conditions I have in parentheses